extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/GrpcServerProcessor.java

```diff
@@ -546,6 +548,18 @@ void configureMetrics(GrpcBuildTimeConfig configuration, Optional<MetricsCapabil
         }
     }
 
+    @BuildStep
+    BeanArchivePredicateBuildItem additionalBeanArchives() {
+        return new BeanArchivePredicateBuildItem(new Predicate<ApplicationArchive>() {
+
+            @Override
+            public boolean test(ApplicationArchive archive) {
+                // Every archive that contains a generated implementor of MutinyBean is considered a bean archive
+                return !archive.getIndex().getKnownDirectImplementors(GrpcDotNames.MUTINY_BEAN).isEmpty();
```

Comment:
is `archive.getIndex()` always `!= null`?

Reply:
I believe so. But it's a good point that we could be more defensive here...

Reply:
Hm, we don't test the nullability in the `BeanArchiveProcessor` either. So it's probably OK.